typing/includecore.ml

```diff
@@ -174,10 +174,22 @@ type variant_change =
   (Types.constructor_declaration as 'l, 'l, constructor_mismatch)
     Diffing_with_keys.change
 
+type type_kind =
+  | Kind_abstract
+  | Kind_record
+  | Kind_variant
+  | Kind_open
+
+let of_kind = function
+  | Type_abstract -> Kind_abstract
+  | Type_record (_, _) -> Kind_record
+  | Type_variant (_, _) -> Kind_variant
+  | Type_open -> Kind_open
```

Comment:
I introduced `type_kind` here to avoid having to put `('lbl, 'cstr) Types.type_kind` into the `Kind` constructor for `type_mismatch`. Using `type_kind` there instead of `('lbl, 'cstr) Types.type_kind` just seems simpler, but it does require introducing this additional `type_kind` type that's really only used in one place.

Reply:
It sounds like you could use  `Types.type_decl_kind`?

Reply:
At the same time, I agree that it is a good practice for the error data types to contain as little information as needed. 
Ideally, I would like all error data types to be serializable in a machine readable format.

Reply:
Good point, `Types.type_decl_kind` is a better candidate than `Types.type_kind`. But it seems like there is no strong objection to introducing an even simpler type in `Includecore`, so I'll go with that for now.